includes/class-wc-payments-webhook-processing-service.php

```diff
@@ -361,6 +361,15 @@ private function process_webhook_payment_intent_succeeded( $event_body ) {
 		$charges_data  = $this->read_webhook_property( $event_charges, 'data' );
 		$charge_id     = $this->read_webhook_property( $charges_data[0], 'id' );
 
+		if ( ! $order ) {
```

Comment:
I would move order fetching at the begging of the function, and use this if statement right after it to prevent getting all this data for nothing if the order is false